### PR TITLE
Add test for search on choice elements

### PR DESCRIPTION
**Pull request by Jasopaum**
Add test for search on choice elements

### CODE REVIEW

**Review by simonvadee — COMMENTED** [review]
can you target the branch of fhirpath implementing this ?

File: fhirstore/fhirstore.py [hunk 1]
```diff
@@ -335,6 +335,8 @@ def search(
             return ValidationError(e).format(as_json)
         except fhirpath.exceptions.ValidationError as e:
             return ValidationError(str(e)).format(as_json)
+        except NotImplementedError as e:
+            return NotSupportedError(str(e)).format(as_json)
```
Comment (simonvadee) [inline fhirstore/fhirstore.py:339]: :+1:

File: test/test_search.py [hunk 2]
```diff
@@ -964,6 +964,18 @@ def test_searchparam_type_uri_above(store: FHIRStore, index_resources):
...
+@pytest.mark.resources("servicerequest-example-lipid.json", "servicerequest-example-myringotomy.json")
+def test_search_on_choice_elements(store: FHIRStore, index_resources):
+    """ """
+    # occurrenceDateTime
+    result = store.search("ServiceRequest", query_string="occurrence=2013-05-02T16:16:00-07:00")
+    assert result.total == 1
+
+    # occurrencePeriod
+    result = store.search("ServiceRequest", query_string="occurrence=2014-03-12")
+    assert result.total == 1
```
Comment (simonvadee) [inline test/test_search.py:981]: usually I like to add a test where _no result_ is found, just to make sure it does not always return a result

**Review by simonvadee — APPROVED** [review]

File: test/test_search.py (hunk 2 above)
Comment (Jasopaum) [reply test/test_search.py:981]: 👍

**Review by simonvadee — APPROVED** [review]
🚀

**Review by simonvadee — APPROVED** [review]

**Pull request merged**